lib/src/workspace.rs

```diff
@@ -460,17 +460,70 @@ impl<'a> LockedWorkspace<'a> {
     }
 }
 
+pub trait WorkspaceLoaderFactory {
+    fn create(&self, workspace_root: &Path)
+        -> Result<Box<dyn WorkspaceLoader>, WorkspaceLoadError>;
+}
+
+pub fn get_working_copy_factory<'a>(
+    workspace_loader: &dyn WorkspaceLoader,
+    working_copy_factories: &'a WorkingCopyFactories,
+) -> Result<&'a dyn WorkingCopyFactory, StoreLoadError> {
```

Comment:
It's not possible to elide this lifetime?

Reply:
I don't think so, because there are two input lifetimes.